### ISSUE TITLE
Installation of plugins docker

### ISSUE THREAD

**Issue by SchrodingersGat** [inventree/InvenTree#3969]
### Discussed in https://github.com/inventree/InvenTree/discussions/3954

<div type='discussions-op-text'>

<sup>Originally posted by **martin323232** November 19, 2022</sup>
Trying to install Brother plugin.
inventree-brother-plugin is inside plugins.txt file

In terminal: docker-compose run inventree-server invoke install

![Screenshot from 2022-11-18 16-31-34](https://user-images.githubusercontent.com/6439801/202743173-1c782d69-f74b-41a7-a300-90f75c5f17c7.png)
![Screenshot from 2022-11-18 16-32-12](https://user-images.githubusercontent.com/6439801/202743182-a85b0fc8-1bf1-4106-8560-e107460a06ad.png)

Tried installing from docker terminal after installing pip.

How do you install from from webpage, "Install Plugin"?
</div>

**Comment by SchrodingersGat** [comment]
@matmair I can replicate this issue:

1. Add `inventree-brother-plugin` to *plugins.txt*
2. Run `docker compose run inventree-server invoke install` 
3. Launch server with `docker compose up -d`
4. Brother plugin is *not* installed

The reason is that 2) installs the plugin into the particular *instance* of the container, but other containers do not respect this. The "development" version keeps everything in a virtual environment which is stored in the external volume, so this issues is avoided as the installation is "persistent".

Basically any plugins installed in the production docker image will not "persist" across different runs.

Also, a secondary issue, adding a plugin via the web interface does not appear to actually add it to "plugins.txt"

**Comment by SchrodingersGat** [comment]
> Also, a secondary issue, adding a plugin via the web interface does not appear to actually add it to "plugins.txt"

Actually, got this to work, did not realize that you had to leave the URL blank if installing via PyPi package name

**Comment by SchrodingersGat** [comment]
Mostly likely due to the "PLUGIN_ON_STARTUP" setting being set to False.

Need to add some documentation around this.

**Comment by matmair** [comment]
> > Also, a secondary issue, adding a plugin via the web interface does not appear to actually add it to "plugins.txt"
> 
> Actually, got this to work, did not realize that you had to leave the URL blank if installing via PyPi package name

There are cases where a URL is needed. Should the validation be clearer?
In general, I think a new interface (hint hint) could really benefit from tooltips etc

**Comment by SchrodingersGat** [comment]
> In general, I think a new interface (hint hint) could really benefit from tooltips etc

wink
